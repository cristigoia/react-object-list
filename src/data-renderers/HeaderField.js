import React from 'react'
import PropTypes from 'prop-types'

/**
 * Represents the text and sorting direction arrow of a TableHeader
 * one or more of these may exist within a single TableHeader
 */
export default class HeaderField extends React.Component {
  static propTypes = {
    /** the sort key used for this particular header item */
    dataKey: PropTypes.string,
    /** the sorting currently used for the data in the */
    activeSort: PropTypes.oneOf([true, false, null]),
    /** whether this header field can be sorted or not */
    sortable: PropTypes.bool,
    /** callback function passed down to determine the sorting of data loaded */
    updateSorting: PropTypes.func,
    /** class names that can be used for additional styling of the component */
    className: PropTypes.string,
    /** the text displayed within the component */
    header: PropTypes.oneOfType([PropTypes.string, PropTypes.func]).isRequired,
  }

  static defaultProps = {
    activeSort: null,
  }

  /**
   * Determines whether to sort the particular column (Field) in
   * ascending or descending order, passing the sorting key back
   * to the ApiList
   *
   * @param {MouseEvent} event - event triggered from an onClick
   */
  handleClick = (event) => {
    if (this.props.dataKey) {
      if (this.props.activeSort) {
        this.props.updateSorting(this.props.dataKey, false)
      } else {
        this.props.updateSorting(this.props.dataKey, true)
      }
    }
  }

  _renderSortIcon = () => {
    switch (this.props.activeSort) {
      case true:
        return (
          <span className="sort-direction text-primary">
            &nbsp;<i className="fa fa-caret-up" aria-hidden="true" />
          </span>
        )
      case false:
        return (
          <span className="sort-direction text-primary">
            &nbsp;<i className="fa fa-caret-down" aria-hidden="true" />
          </span>
        )
      default:
        return undefined
    }
  }

  _renderHeader = () => {
    switch (typeof this.props.header) {
      case 'function':
        return <this.props.header />
      case 'string':
        return this.props.header
      default:
        return ''
    }
  }

  render() {
    if (this.props.sortable) {
      return (
        <div
          role={this.props.dataKey ? 'button' : '' /* role[button] adds cursor: pointer */}
          onClick={this.handleClick} className={this.props.className} >
          {this._renderHeader()}{this._renderSortIcon()}
        </div>
      )
    } else {
      return <span>{this._renderHeader()}</span>
    }
  }
}
