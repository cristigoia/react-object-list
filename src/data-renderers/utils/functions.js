/**
 * Gets the value of the cell from the row and keys provided
 * @param  {Object} row  Row supplied
 * @param  {Array|string} keys List of keys used to access the value
 * @return {(Object|string|number)}      Value from row, or row itself if no keys
 */
const getValueFromAccessor = (row, keys) => {
  if (!Array.isArray(keys)) keys = keys.split('.')
  let value = row
  const cloneKeys = [...keys]
  while (value && cloneKeys.length) {
    const thisKey = cloneKeys.shift()
    if (thisKey in value) {
      value = value[thisKey]
    }
  }
  return value
}

export {
  getValueFromAccessor,
}
