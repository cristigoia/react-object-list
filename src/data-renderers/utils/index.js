export {
  getValueFromAccessor,
  handleRowClick,
} from './functions'
