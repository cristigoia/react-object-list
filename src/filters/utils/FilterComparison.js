import React from 'react'
import PropTypes from 'prop-types'
import Select from 'react-select'

class FilterComparison extends React.Component {
  static propTypes = {
    options: PropTypes.arrayOf(PropTypes.shape({
      label: PropTypes.string,
      value: Select.propTypes.value,
    })),
    value: Select.propTypes.value,
    onChange: PropTypes.func.isRequired,
  }

  render() {
    const {options, value} = this.props
    return (
      <div className="objectlist-current-filter__filter-comparison">
        <Select
          options={options}
          value={value}
          onChange={newValue => this.props.onChange(newValue.value)}
          clearable={false}
        />
      </div>
    )
  }
}

export default FilterComparison
