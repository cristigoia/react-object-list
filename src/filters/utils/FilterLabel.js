import React from 'react'
import PropTypes from 'prop-types'

/**
 * Component to render label for filters
 */
class FilterLabel extends React.Component {
  static propTypes = {
    /** Text to render in the label */
    label: PropTypes.string.isRequired,
  }
  render() {
    return (
      <label className="objectlist-current-filter__filter-name">{this.props.label}:</label>
    )
  }
}

export default FilterLabel
