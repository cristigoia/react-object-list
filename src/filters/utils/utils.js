const possibleQueryStringLookupExpressions = ['exact', 'lt', 'gt', 'gte', 'lte']

export {
  possibleQueryStringLookupExpressions,
}
