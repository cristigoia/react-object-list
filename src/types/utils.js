function floatPrecision(floatValue, precision) {
  floatValue = parseFloat(floatValue)
  if (isNaN(floatValue)) { return parseFloat('0').toFixed(precision) } else {
    const power = Math.pow(10, precision)
    floatValue = (Math.round(floatValue * power) / power).toFixed(precision)
    return floatValue.toString()
  }
}

function commaNumber(numberString) {
  const parts = numberString.toString().split('.')
  parts[0] = parts[0].replace(/\B(?=(\d{3})+(?!\d))/g, ',')
  return parts.join('.')
}
/**
 * Converts a float to stringed currency representation
 * @param  {number} floatNumber Amount to convert
 * @return {string}             Dollar string
 */
export function currency(floatNumber) {
  return `${floatNumber < 0 ? '-' : ''}$${commaNumber(floatPrecision(Math.abs(floatNumber), 2))}`
}
