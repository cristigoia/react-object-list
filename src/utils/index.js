export * from './constants'
export * from './proptypes'
export * from './functions'
export loadingSpinner from './loading'
