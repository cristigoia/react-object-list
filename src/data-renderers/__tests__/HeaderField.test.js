import React from 'react'
import { snapshotTest } from 'utils/tests'
import HeaderField from '../HeaderField'

describe('header field', () => {
  describe('snapshopt', () => {
    const props = {
      key: 'header_sort_key',
      header: 'Header Text',
      sortable: true,
    }
    it('renders correctly', () => {
      snapshotTest(<HeaderField {...props} />)
    })
  })
})
