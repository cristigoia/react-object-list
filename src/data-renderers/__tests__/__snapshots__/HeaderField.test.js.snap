// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header field snapshopt renders correctly 1`] = `
<div
  className={undefined}
  onClick={[Function]}
  role=""
>
  Header Text
</div>
`;
